cpp/libs/src/asiodnp3/tls/MasterTLSServer.cpp

```diff
@@ -71,28 +71,32 @@ bool MasterTLSServer::AcceptConnection(uint64_t sessionid, const asio::ip::tcp::
 bool MasterTLSServer::VerifyCallback(uint64_t sessionid, bool preverified, asio::ssl::verify_context& ctx)
 {
 	int depth = X509_STORE_CTX_get_error_depth(ctx.native_handle());
+	
+	// lookup the subject name
+	X509* cert = X509_STORE_CTX_get_current_cert(ctx.native_handle());
+	char subjectName[512];
+	X509_NAME_oneline(X509_get_subject_name(cert), subjectName, 512);
```

Comment:
Should the X509_NAME returned from X509_get_subject_name be freed? The OpenSSL docs suggest it shouldn't:

> X509_get_subject_name() returns the subject name of certificate x. The returned value is an internal pointer which MUST NOT be freed.

... however I've seen a few projects that do free it:

https://github.com/nginx/nginx/blob/3c44339bfed78c9ad3549edf2b7408c15d3ae51c/src/event/ngx_event_openssl.c#L782